### ISSUE TITLE
consume method silently ignores amounts bigger than the total capacity

### ISSUE THREAD

**Issue by gcscaglia** [vladimir-bukhtoyarov/bucket4j#20]
Consider the following snippet:

```java
Bucket bucket = Bucket4j.builder()
        .withLimitedBandwidth(1, Duration.ofSeconds(1))
        .build();

for (int i = 0; i < 5; i++) {
    bucket.consume(5);
}
```

You did expect the loop to take 5 * 5 = 25 seconds. But it returns instantly: No exceptions, no wait.

While not supporting this explicitly is an option, an appropriate exception should be thrown; silently retuning seems like a bug to me.

**Comment by vladimir-bukhtoyarov** [comment]
Looks like there are two bugs:
- Firstly in design of API, javadocs says nothing about what will be happen in case of number of tokens to consume is greater than bucket capacity.
- Secondly bug in implementation which does not check this corner case.

What is impact of this issue for you? As quick fix, I can throw IllegalArgumentException. Supporting of long waiting also is planned, but it cannot be delivered fast due to time limitations which I currently have.

**Comment by gcscaglia** [comment]
To me, the most convenient would be the long wait support, but it's no big deal if there isn't; In my case I can workaround it easily by scaling the values.

The bigger problem was the time it took me to realize this was the source of my problems; Hence the exception suggestion.

**Comment by vladimir-bukhtoyarov** [comment]
Ok, I will fix the issue in two steps, the first will be very simple, just clarification of javadocs and raising of IllegalArgumentException.

**Comment by vladimir-bukhtoyarov** [comment]
Done, will be released in scope of "2.0".
